adr/0002-keycloak-gitlab.md

```diff
@@ -0,0 +1,165 @@
+- Potential for Misconfiguration: Misconfigurations can lead to unauthorized access or the accidental removal of access rights, impacting operations and security.
+- Complexity: Integrating SCIM with existing systems can be complex.
+
+#### Manual Deprovisioning
+**Pros:**
+Direct Control Over the Process: This gives administrators complete oversight of when and how users are deprovisioned, ensuring no automated system can mistakenly alter access rights.
+- Simple to Implement: It doesn't require introducing new tools or technologies beyond what's already in use, making it straightforward to execute.
+
+**Cons:**
+Labor-Intensive: Each de-provisioning event requires manual action, which can be time-consuming, especially for organizations with a high volume of user changes.
+- Prone to Delays: The manual process is susceptible to delays, which can lead to unauthorized access if users without access retain it due to oversight or backlog.
+- Not Scalable:
+
+#### Custom Code
+**Pros:**
+- Flexibility: You can tailor the de-provisioning process to your exact needs, ensuring it fits your organization perfectly.    
+
+**Cons:**
+- More Work: You'll have to write and maintain the code yourself, which can be a lot of work.
+
+#### GitLab Group Integration
+
+Within GitLab, groups are often used to control users' access to groups of repositories. Groups can be manually configured by a GitLab group owner, manually by a GitLab instance admin or automatically from SAML groups. GitLab does not currently support SCIM groups.
+
+#### Automated Group Management via Keycloak
+**Pros:**
+- Easier Access Management: Linking GitLab with Keycloak groups means when someone's group membership changes in Keycloak, it updates in GitLab too. This keeps access rights tight without extra effort.
+- Automatic Updates: Group memberships are updated automatically, saving a lot of manual work and ensuring no one is accidentally left in or out of a group.
+- Better Security: With automatic updates, you can be sure that everyone has the access they need and nothing more, keeping your projects and data safer.
+
+**Cons:**
+- Requires administering groups from Keycloak instead of GitLab. This may be more complex and not all users may have Keycloak group admin access. (However, this may scale better to large numbers of groups.)
+- Requires Careful Configuration: Setting up Keycloak to work correctly with GitLab requires careful planning and configuration.
+- Learning Curve: more complex to administer groups from Keycloak instead of GitLab. Another admin tool to learn and some complexity from not using GitLab directly.
+
+#### Manual Group Management
+**Pros:**
+- Simple to set up, likely works well for small GitLab installations
+- You're in Charge: Hand-managing groups means you make all the calls about who's in what group, giving you tight control over project access.
```

Comment:
Stylistic thing, but usually design docs use impersonal wording or generically `we`/`us`

Reply:
👍